### DISCUSSION TITLE
Test Engine build error: Unresolved external symbols

### DISCUSSION THREAD

**Discussion by Tymoteusz-Mroczkowski** [ocornut/imgui#7319] [New Users Build/Link/Run issues ONLY!]
I'm trying to add the Test Engine to my application but no matter what I do the compiler (Visual Studio) is failing with 4 LNK2019 errors.
error LNK2019: unresolved external symbol "void __cdecl ImGuiTestEngineHook_ItemAdd
error LNK2019: unresolved external symbol "void __cdecl ImGuiTestEngineHook_ItemInfo
error LNK2019: unresolved external symbol "void __cdecl ImGuiTestEngineHook_Log
error LNK2019: unresolved external symbol "char const * __cdecl ImGuiTestEngine_FindItemDebugLabel

As instructed in the wiki I am including imgui_te_imconfig.h in my imconfig.h, I tried to put test-engine in different locations relative to Imgui but with no changes to the errors.

The Imgui version I'm using is 1.90.1 with SDL2 platform and opengl2 renderer backends.

**Comment by ocornut** [comment]
This is the wrong repository to ask. But anyhow unresolved external almost always mean you are not compiling/linking with test engine files.

**Comment by Tymoteusz-Mroczkowski** [comment]
Thank you, I fixed the problem, I had to add the test-engine files in the same add_library() as imgui in my CMakeLists.txt and not a separate one.

**Reply by ocornut** [reply]
Glad to hear, hope the test engine will be useful to you ! Don't hesitate to ask questions in the https://github.com/ocornut/imgui_test_engine repository.
